### PR TITLE
Fix campaign reports saving incorrect results when price changes

### PR DESCRIPTION
**Pull request by dwick**
👓 @zeantsoi 

same as this issue: https://github.com/reddit/reddit-plugin-adzerk/pull/146 but for the lifetime reports instead of the daily ones.

### CODE REVIEW

**Comment by dwick** [comment]
issue: https://reddit.atlassian.net/browse/ADS-743

File: reddit_adzerk/adzerkreporting.py [hunk 1]
```diff
@@ -355,13 +356,20 @@ def _process_lifetime_campaign_reports(campaigns, report_id, queued_date):
...
-            campaign.adserver_spent_pennies = int(spent * 100)
-            campaign.adserver_impressions = impressions
-            campaign.adserver_clicks = clicks
-            campaign.last_lifetime_report = report_id
-            campaign.last_lifetime_report_run = queued_date
+            # if the price changes there may be more than 1 record for a single campaign.
+            campaign_values = campaign_results[campaign]
+            campaign_values["impressions"] = campaign_values["impressions"] + impressions
+            campaign_values["clicks"] = campaign_values["clicks"] + clicks
+            campaign_values["spent_pennies"] = campaign_values["spent_pennies"] + (spent * 100.)
```
Comment (zeantsoi) [inline reddit_adzerk/adzerkreporting.py]: What do we do with `campaign_values`?

File: reddit_adzerk/adzerkreporting.py [hunk 2]
```diff
@@ -335,6 +335,7 @@ def _process_lifetime_campaign_reports(campaigns, report_id, queued_date):
     report_records = report_result.get("Records", None)
     campaigns_by_fullname = {c._fullname: c for c in campaigns}
 
+    campaign_results = defaultdict(_reporting_factory)
```
Comment (zeantsoi) [inline reddit_adzerk/adzerkreporting.py]: Does this only need to have the function passed, or does it need to be called as `_reporting_factory()`?

File: reddit_adzerk/adzerkreporting.py (hunk 1 above)
Comment (dwick) [inline reddit_adzerk/adzerkreporting.py]: https://github.com/reddit/reddit-plugin-adzerk/pull/241/files#diff-a369a8ee1a6bfac28a8b167eb9f58fa7R365

Comment (zeantsoi) [inline reddit_adzerk/adzerkreporting.py]: Oh, I see. Passed-by-reference. 👍

**Comment by zeantsoi** [comment]
[One comment](https://github.com/reddit/reddit-plugin-adzerk/pull/241/files#r77036390) 💅

File: reddit_adzerk/adzerkreporting.py (hunk 2 above)
Comment (dwick) [inline reddit_adzerk/adzerkreporting.py]: no, when a default dict receives a function it calls that when the key is missing upon index.

basically:
def defaultdict(fn):
   dict = {}
   return lambda key: dict[key] if key in dict else fn()

**Comment by dwick** [comment]
💇

**Comment by zeantsoi** [comment]
🐟

**Pull request merged**